src/FiltrationConditionComponent/FiltrationConditionComponent.tsx

```diff
@@ -21,81 +25,80 @@ export class FiltrationConditionComponent extends React.Component<
 
 	public state = {
-		offsetTop: 0
+		offsetTop: 0,
+		showPopover: false,
+		popoversChildren: [],
+		popoverFilterAction: false
 	};
-	private popoverElement: Element = document.createElement("div");
+	private observer: IntersectionObserver;
 
 	public componentDidMount() {
-		window.addEventListener("scroll", this.scrollWindowsHeight);
-		this.handleMoveUpParentPopupSegment();
-	}
+		const refCondition = this.refCondition.current;
 
-	public componentWillUnmount() {
-		const refContent = this.refContent.current;
+		if (refCondition) {
+			requestAnimationFrame(() => {
+				const { top } = refCondition.getBoundingClientRect();
 
-		window.removeEventListener("scroll", this.scrollWindowsHeight);
-		window.removeEventListener("load", this.handleMoveUpParentPopupSegment);
+				this.headerOffsetTop = top - RANGE_OFFSET_TOP;
 
-		if (refContent) {
-			if (refContent.parentElement) {
-				refContent.parentElement.parentElement!.removeChild(
-					this.popoverElement
+				const observerOptions = {
+					rootMargin: `-${Math.abs(
+						this.headerOffsetTop
+					)}px 0px 0px 0px`,
+					threshold: 1.0
+				};
+
+				this.observer = new IntersectionObserver(
+					this.checkCollapsed,
+					observerOptions
 				);
-			}
+				this.observer.observe(refCondition);
+			});
 		}
 	}
 
-	public scrollWindowsHeight = () => {
-		const { offsetTop } = this.state;
-		const { onConditionStateToggle, state } = this.props;
-		const refContent = this.refContent.current;
-
-		if (refContent) {
-			const { top } = refContent.getBoundingClientRect();
-			if (offsetTop === 0) {
-				this.setState({
-					offsetTop: top
-				});
-			}
-
-			if (
-				top > 0 &&
-				window.innerHeight >= top &&
-				offsetTop - RANGE_OFFSET_TOP > top &&
-				onConditionStateToggle &&
-				state === "edit"
-			) {
-				onConditionStateToggle();
-			}
-		}
-	};
+	public componentWillUnmount() {
+		const refCondition = this.refCondition.current;
 
-	public handleMoveUpParentPopupSegment = () => {
-		const refContent = this.refContent.current;
+		if (refCondition) {
+			this.observer.disconnect();
+		}
+	}
 
-		if (refContent) {
-			const popover = refContent.querySelector(
-				".kit-segment-button-expand__popover"
-			);
+	public checkCollapsed = (entries: IntersectionObserverEntry[]): void => {
+		const refCondition = this.refCondition.current;
+		const { onConditionStateToggle, state } = this.props;
 
-			if (popover) {
-				this.popoverElement = popover;
-				popover.remove();
-			}
+		const entry = entries.find(
+			({ target }) => target === refCondition
+		) as IntersectionObserverEntry;
 
-			if (refContent.parentElement) {
-				refContent.parentElement.parentElement!.appendChild(
-					this.popoverElement
-				);
-			}
+		if (entry && state === "edit" && !entry.isIntersecting) {
+			onConditionStateToggle();
 		}
+	};
 
-		window.addEventListener("load", this.handleMoveUpParentPopupSegment);
+	public renderPopover = (
+		children: React.ReactNode,
+		filterAction: JSX.Element
```

Comment:
Вроде уже `React.ReactNode` это
```suggestion
		filterAction: React.ReactNode
```